manipmongo/manipulator.go

```diff
@@ -62,11 +62,14 @@ func New(url string, db string, options ...Option) (manipulate.TransactionalMani
 
 	if cfg.tlsConfig != nil {
 		dialInfo.DialServer = func(addr *mgo.ServerAddr) (net.Conn, error) {
-			return tls.Dial("tcp", addr.String(), cfg.tlsConfig)
+			d := net.Dialer{Timeout: dialInfo.Timeout}
+			return tls.DialWithDialer(&d, "tcp", addr.String(), cfg.tlsConfig)
```

Comment:
Instead of creating a variable, we could write everything within the return with formatting:
```golang
return tls.DialWithDialer(
   &net.Dialer{
       Timeout: dialInfo.Timeout},
   },
   "tcp",
   addr.String(),
   cfg.tlsConfig,
)
```